app/src/main/kotlin/org/fossify/phone/adapters/RecentCallsAdapter.kt

```diff
@@ -480,16 +489,58 @@ class RecentCallsAdapter(
                     setTextSize(TypedValue.COMPLEX_UNIT_PX, currentFontSize * 0.8f)
                 }
 
+                itemRecentsDateTimeDurationSeparator.apply {
+                    text = " • "
+                    setTextSize(TypedValue.COMPLEX_UNIT_PX, currentFontSize * 0.8f)
+                    setTextColor(textColor)
+                    if (shouldShowDuration) {
+                        beVisible()
+                    } else {
+                        beInvisible()
+                    }
+                }
+
                 itemRecentsDuration.apply {
-                    text = call.duration.getFormattedDuration()
+                    text = context.formatSecondsToShortTimeString(call.duration)
                     setTextColor(textColor)
-                    beVisibleIf(call.type != Calls.MISSED_TYPE && call.type != Calls.REJECTED_TYPE && call.duration > 0)
+                    if (shouldShowDuration) {
+                        beVisible()
+                    } else {
+                        beInvisible()
+                    }
```

Comment:
We don't need to make them 'invisible' now that the location text is always aligned to the end. 'Gone' is good.

```suggestion
                    beVisibleIf(shouldShowDuration)
```